src/content/docs/en/guides/typescript.mdx

```diff
@@ -34,6 +34,44 @@ Additionally, our templates include an `env.d.ts` file inside the `src` folder t
 /// <reference types="astro/client" />
 ```
 
+If you are not using VSCode, you can install the [Astro TypeScript plugin](https://www.npmjs.com/package/@astrojs/ts-plugin) to support importing `.astro` files from `.ts` files (which can be useful for re-exporting). 
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+  ```shell
+  npm install @astrojs/ts-plugin
+  ```
+  </Fragment>
+  <Fragment slot="pnpm">
+  ```shell
+  pnpm install @astrojs/ts-plugin
+  ```
+  </Fragment>
+  <Fragment slot="yarn">
+  ```shell
+  yarn add @astrojs/ts-plugin
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+Then, add the following to your `tsconfig.json`:
+
+```json title="tsconfig.json"
+  "compilerOptions": {
+    "plugins": [
+      {
+        "name": "@astrojs/ts-plugin"
+      },
+    ],
+    "jsx": "react",
+```
+
+To check that the plugin is working, create a `.ts` file and import an Astro component into it. You should have no warning messages from your editor.
+
+:::note
+The `jsx` setting specifies what JSX code is generated. You may have to change it depending on the particular framework you are using. For example, SolidJS users may prefer the "preserve" option. For plain Astro projects, "react" works well. Valid values are "preserve", "react", "react-jsx", "react-jsdev" and "react-native".
+:::
```

Comment:
```suggestion
```
I don't think this section is necessary. Most of these values don't help in Astro, and `jsx: preserve` always works.